app/controllers/account_activations_controller.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class AccountActivationsController < ApplicationController
+  def edit
+    if user && !user.activated? && user.authenticated?(:activation, params[:id])
+      activate_and_log_in_user
+      flash[:success] = 'Account activated!'
+      redirect_to user
+    else
+      flash[:danger] = 'Invalid activation link'
+      redirect_to root_url
+    end
+  end
+
+  private
+
+  def user
+    @user = User.find_by(email: params[:email])
+  end
```

Comment:
これだと、`user`を呼ぶたびにインスタンスが生成されて非効率なので、メモ化しましょう

```suggestion
  def user
    @user ||= User.find_by(email: params[:email])
  end
```

Reply:
対応しました。